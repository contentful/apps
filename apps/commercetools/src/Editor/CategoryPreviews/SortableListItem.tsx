import React from "react";
import { SortableElement, SortableHandle } from "react-sortable-hoc";
import { css } from "emotion";
import {
  Card,
  CardDragHandle as FormaCardDragHandle,
  Heading,
  Icon,
  IconButton,
  Subheading,
  Tag,
  Typography
} from "@contentful/forma-36-react-components";
import tokens from "@contentful/forma-36-tokens";
import { Category } from "../../interfaces";

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: "flex",
    padding: 0,
    position: "relative",
    ":not(:first-of-type)": css({
      marginTop: tokens.spacingXs
    })
  }),
  dragHandle: css({
    height: "auto"
  }),
  actions: css({
    position: "absolute",
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: "inline-block",
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`
      }),
      "&:hover": {
        svg: css({
          fill: tokens.colorContrastDark
        })
      }
    })
  }),
  description: css({
    padding: tokens.spacingM,
    flex: "1 0 auto",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center"
  }),
  name: (name?: string) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: tokens.spacing2Xs,
      ...(name && { textTransform: "capitalize" })
    }),
  slug: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0
  })
};

const CardDragHandle = SortableHandle(() => (
  <FormaCardDragHandle className={styles.dragHandle}>
    Reorder category
  </FormaCardDragHandle>
));

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    const categoryIsMissing = !category.name;

    return (
      <Card className={styles.card}>
        <>
          {isSortable && <CardDragHandle />}
          <section className={styles.description}>
            <Typography>
              <Heading className={styles.name(category.name)}>
                {categoryIsMissing ? category.id : category.name}
              </Heading>
              {categoryIsMissing ? (
                <Tag tagType="negative">Category missing</Tag>
              ) : (
                <Subheading className={styles.slug}>{category.slug}</Subheading>
              )}
            </Typography>
          </section>
        </>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a
                target="_blank"
                rel="noopener noreferrer"
                href={category.externalLink}
              >
                <Icon icon="ExternalLink" color="muted" />
              </a>
            )}
            <IconButton
              label="Delete"
              iconProps={{ icon: "Close" }}
              {...{
                buttonType: "muted",
                onClick: onDelete
              }}
            />
          </div>
        )}
      </Card>
    );
  }
);
