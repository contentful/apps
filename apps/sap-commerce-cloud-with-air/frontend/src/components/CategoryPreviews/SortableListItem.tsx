import { ReactElement } from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import { Badge, Card, Heading, IconButton, Subheading } from '@contentful/f36-components';
import { CloseIcon, ExternalLinkIcon } from '@contentful/f36-icons';
import tokens from '@contentful/forma-36-tokens';
import { Category } from '../../interfaces';

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: 'flex',
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  cardInner: css({
    display: 'flex',
  }),
  dragHandle: css({
    height: 'auto',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorBlack,
        }),
      },
    }),
  }),
  description: css({
    padding: tokens.spacingM,
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (category: Category) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: category.isMissing || !category.name ? 0 : tokens.spacing2Xs,
      ...(category.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.gray500,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  slug: css({
    color: tokens.colorBlack,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
};

const CardDragHandle = SortableHandle(({ drag }: { drag: ReactElement }) => <>{drag}</>);

function getCategoryIdentifier(category: Category) {
  return category.slug.length ? category.slug : category.id;
}

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    return (
      <Card
        className={styles.card}
        withDragHandle
        dragHandleRender={isSortable ? ({ drag }) => <CardDragHandle drag={drag} /> : undefined}>
        <div className={styles.cardInner}>
          <section className={styles.description}>
            <>
              <Heading className={styles.heading(category)}>
                {category.isMissing || !category.name
                  ? getCategoryIdentifier(category)
                  : category.name}
              </Heading>
              {category.isMissing && <Badge variant="negative">Category missing</Badge>}
              {!category.isMissing && category.name && (
                <Subheading className={styles.subheading}>
                  {getCategoryIdentifier(category)}
                </Subheading>
              )}
            </>
          </section>
        </div>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={category.externalLink}>
                <ExternalLinkIcon color="muted" />
              </a>
            )}
            <IconButton
              aria-label="Delete"
              icon={<CloseIcon />}
              variant="transparent"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  },
);
