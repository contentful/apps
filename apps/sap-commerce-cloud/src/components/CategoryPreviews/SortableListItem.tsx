import React from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import {
  Card,
  CardDragHandle as FormaCardDragHandle,
  Heading,
  Icon,
  IconButton,
  Subheading,
  Tag,
  Typography,
} from '@contentful/forma-36-react-components';
import tokens from '@contentful/forma-36-tokens';
import { Category } from '../../interfaces';

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: 'flex',
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  dragHandle: css({
    height: 'auto',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorContrastDark,
        }),
      },
    }),
  }),
  description: css({
    padding: tokens.spacingM,
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (category: Category) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: category.isMissing || !category.name ? 0 : tokens.spacing2Xs,
      ...(category.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  slug: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
};

const CardDragHandle = SortableHandle(() => (
  <FormaCardDragHandle className={styles.dragHandle}>Reorder category</FormaCardDragHandle>
));

function getCategoryIdentifier(category: Category) {
  return category.slug.length ? category.slug : category.id;
}

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    return (
      <Card className={styles.card}>
        <>
          {isSortable && <CardDragHandle />}
          <section className={styles.description}>
            <Typography>
              <Heading className={styles.heading(category)}>
                {category.isMissing || !category.name
                  ? getCategoryIdentifier(category)
                  : category.name}
              </Heading>
              {category.isMissing && <Tag tagType="negative">Category missing</Tag>}
              {!category.isMissing && category.name && (
                <Subheading className={styles.subheading}>
                  {getCategoryIdentifier(category)}
                </Subheading>
              )}
            </Typography>
          </section>
        </>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={category.externalLink}>
                <Icon icon="ExternalLink" color="muted" />
              </a>
            )}
            <IconButton
              label="Delete"
              iconProps={{ icon: 'Close' }}
              buttonType="muted"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  }
);
