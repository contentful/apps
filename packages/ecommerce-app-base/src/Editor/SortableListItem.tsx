import {
  Badge,
  Card,
  Heading,
  IconButton,
  SkeletonContainer,
  SkeletonImage,
  Subheading,
} from '@contentful/f36-components';
import { AssetIcon, CloseIcon, ErrorCircleIcon, ExternalLinkIcon } from '@contentful/f36-icons';
import tokens from '@contentful/f36-tokens';
import { css } from 'emotion';
import React, { ReactElement, useState } from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { Product } from '../interfaces';

export interface Props {
  product: Product;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
  skuType?: string;
}

const IMAGE_SIZE = 48;

const styles = {
  card: css({
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  cardInner: css({
    display: 'flex',
  }),
  imageWrapper: (imageHasLoaded: boolean) =>
    css({
      width: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      height: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      overflow: 'hidden',
      margin: imageHasLoaded ? tokens.spacingM : 0,
      position: 'relative',
      '> img': css({
        display: 'block',
        height: `${IMAGE_SIZE}px`,
        minWidth: 'auto',
        userSelect: 'none',
        position: 'absolute',
        left: '50%',
        top: '50%',
        transform: 'translate(-50%, -50%)',
      }),
    }),
  dragHandle: css({
    height: 'auto',
    borderBottomLeftRadius: tokens.borderRadiusMedium,
    borderTopLeftRadius: tokens.borderRadiusMedium,
    cursor: 'grab',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorBlack,
        }),
      },
    }),
  }),
  description: css({
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  name: (name?: string) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: tokens.spacing2Xs,
      ...(name && { textTransform: 'capitalize' }),
    }),
  sku: css({
    color: tokens.gray500,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  skeletonImage: css({
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    padding: tokens.spacingM,
  }),
  errorImage: css({
    backgroundColor: tokens.gray100,
    borderRadius: '3px',
    margin: tokens.spacingM,
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    position: 'relative',
    svg: css({
      fill: tokens.gray600,
      width: '100%',
      height: '50%',
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    }),
  }),
};

const CardDragHandle = SortableHandle(({ drag }: { drag: ReactElement }) => <>{drag}</>);

export const SortableListItem = SortableElement<Props>(
  ({ product, disabled, isSortable, onDelete, skuType }: Props) => {
    const [imageHasLoaded, setImageLoaded] = useState(false);
    const [imageHasErrored, setImageHasErrored] = useState(false);
    const productIsMissing = !product.name;

    return (
      <Card
        data-testid="sortable-list-item"
        className={styles.card}
        // @ts-ignore
        withDragHandle
        dragHandleRender={isSortable ? ({ drag }) => <CardDragHandle drag={drag} /> : undefined}
      >
        <div className={styles.cardInner}>
          {!imageHasLoaded && !imageHasErrored && product.image && (
            <SkeletonContainer className={styles.skeletonImage}>
              <SkeletonImage width={IMAGE_SIZE} height={IMAGE_SIZE} />
            </SkeletonContainer>
          )}
          {!product.image || imageHasErrored ? (
            <div className={styles.errorImage}>
              {productIsMissing ? (
                <ErrorCircleIcon testId="error-circle-icon" />
              ) : (
                <AssetIcon testId="asset-icon" />
              )}
            </div>
          ) : (
            <div className={styles.imageWrapper(imageHasLoaded)}>
              <img
                style={{ display: imageHasLoaded ? 'block' : 'none' }}
                onLoad={() => setImageLoaded(true)}
                onError={() => setImageHasErrored(true)}
                src={product.image}
                alt={product.name}
                data-test-id="image"
              />
            </div>
          )}
          <section className={styles.description}>
            <Heading className={styles.name(product.name)}>
              {productIsMissing ? product.sku : product.name}
            </Heading>
            {productIsMissing ? (
              <Badge variant="negative">{skuType ?? 'Product'} missing</Badge>
            ) : (
              <Subheading className={styles.sku}>{product.displaySKU ?? product.sku}</Subheading>
            )}
          </section>
        </div>
        {!disabled && (
          <div className={styles.actions}>
            {product.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={product.externalLink}>
                <ExternalLinkIcon color="muted" />
              </a>
            )}
            <IconButton
              aria-label="Delete"
              icon={<CloseIcon />}
              variant="transparent"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  }
);
