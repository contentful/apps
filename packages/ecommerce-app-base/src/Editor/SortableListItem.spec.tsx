import React from 'react';
import {
  fireEvent,
  configure,
  render,
  cleanup,
  waitFor,
  getByTestId,
} from '@testing-library/react';
import { Props, SortableListItem } from './SortableListItem';
import productPreviews from '../__mocks__/productPreviews';

configure({
  testIdAttribute: 'data-test-id',
});

const defaultProps: Props = {
  product: productPreviews[0],
  disabled: false,
  onDelete: jest.fn(),
  isSortable: false,
};

const renderComponent = (props: Props) => {
  return render(<SortableListItem index={0} {...props} />);
};

jest.mock('react-sortable-hoc', () => ({
  SortableContainer: (x: any) => x,
  SortableElement: (x: any) => x,
  SortableHandle: (x: any) => x,
}));

describe('SortableListItem', () => {
  afterEach(cleanup);

  it('should render successfully', async () => {
    const { getByTestId } = renderComponent(defaultProps);
    const image = getByTestId('image');
    fireEvent(image, new Event('load'));
    expect(image).toHaveStyle('display: block');
  });

  it('should render successfully the sortable variation', () => {
    const { getByTestId } = renderComponent({ ...defaultProps, isSortable: true });
    const image = getByTestId('image');
    fireEvent(image, new Event('load'));
    expect(image).toHaveStyle('display: block');
  });

  it('should render successfully the loading variation', () => {
    const { getByTestId } = renderComponent(defaultProps);
    const image = getByTestId('image');
    expect(image).toHaveStyle('display: none');
  });

  it('should render successfully the error variation for missing image', async () => {
    const { getByTestId } = renderComponent({ ...defaultProps, isSortable: true });
    const image = getByTestId('image');
    fireEvent(image, new Event('error'));
    expect(image).not.toBeInTheDocument();
    expect(getByTestId('asset-icon')).toBeInTheDocument();
  });

  it('should render successfully the error variation for missing product', async () => {
    const { getByTestId } = renderComponent({
      ...defaultProps,
      product: { ...productPreviews[0], name: '' },
    });
    const image = getByTestId('image');
    fireEvent(image, new Event('error'));
    expect(image).not.toBeInTheDocument();
    expect(getByTestId('error-circle-icon')).toBeInTheDocument();
  });
});
