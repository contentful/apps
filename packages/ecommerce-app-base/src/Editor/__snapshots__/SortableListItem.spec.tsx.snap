// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableListItem should render successfully 1`] = `
<div>
  <article
    class="css-1bc2b9n"
    data-test-id="cf-ui-card"
  >
    <div
      class="css-1cjnmwb"
      data-card-part="wrapper"
    >
      <div
        class="css-161cbg"
        data-card-part="content"
      >
        <div
          class="css-14jukvq"
        >
          <img
            alt="Dress Twin-Set rose"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-1.jpg"
            style="display: block;"
          />
        </div>
        <section
          class="css-1phcpb1"
        >
          <h1
            class="css-19gqs9u"
            data-test-id="cf-ui-heading"
          >
            Dress Twin-Set rose
          </h1>
          <h3
            class="css-1dno5oq"
            data-test-id="cf-ui-subheading"
          >
            M0E20130820E90Z
          </h3>
        </section>
        <div
          class="css-mfolfb"
        >
          <a
            href="https://mc.commercetools.com/example-project/products/28a7442d/general"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="css-rm9mk1"
              color="muted"
              data-test-id="cf-ui-icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </a>
          <button
            aria-label="Delete"
            class="css-8mvp1w"
            data-test-id="cf-ui-icon-button"
            type="button"
          >
            <span
              class="css-k008qs"
            >
              <svg
                aria-hidden="true"
                class="css-1jrff5x"
                data-test-id="cf-ui-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`SortableListItem should render successfully the error variation for missing image 1`] = `
<div>
  <article
    class="css-1bc2b9n"
    data-test-id="cf-ui-card"
  >
    <div
      class="css-1cjnmwb"
      data-card-part="wrapper"
    >
      <div
        class="css-161cbg"
        data-card-part="content"
      >
        <div
          class="css-6t06np"
        >
          <svg
            aria-hidden="true"
            class="css-rm9mk1"
            data-test-id="cf-ui-icon"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <g>
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
              <path
                d="M14.14 11.86l-3 3.87L9 13.14 6 17h12l-3.86-5.14z"
              />
            </g>
          </svg>
        </div>
        <section
          class="css-1phcpb1"
        >
          <h1
            class="css-19gqs9u"
            data-test-id="cf-ui-heading"
          >
            Dress Twin-Set rose
          </h1>
          <h3
            class="css-1dno5oq"
            data-test-id="cf-ui-subheading"
          >
            M0E20130820E90Z
          </h3>
        </section>
        <div
          class="css-mfolfb"
        >
          <a
            href="https://mc.commercetools.com/example-project/products/28a7442d/general"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="css-rm9mk1"
              color="muted"
              data-test-id="cf-ui-icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </a>
          <button
            aria-label="Delete"
            class="css-8mvp1w"
            data-test-id="cf-ui-icon-button"
            type="button"
          >
            <span
              class="css-k008qs"
            >
              <svg
                aria-hidden="true"
                class="css-1jrff5x"
                data-test-id="cf-ui-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`SortableListItem should render successfully the error variation for missing product 1`] = `
<div>
  <article
    class="css-1bc2b9n"
    data-test-id="cf-ui-card"
  >
    <div
      class="css-1cjnmwb"
      data-card-part="wrapper"
    >
      <div
        class="css-161cbg"
        data-card-part="content"
      >
        <div
          class="css-6t06np"
        >
          <svg
            aria-hidden="true"
            class="css-rm9mk1"
            data-test-id="cf-ui-icon"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
        </div>
        <section
          class="css-1phcpb1"
        >
          <h1
            class="css-1wlxn5b"
            data-test-id="cf-ui-heading"
          >
            M0E20130820E90Z
          </h1>
          <div
            class="css-13doom3"
            data-test-id="cf-ui-badge"
          >
            Product
             missing
          </div>
        </section>
        <div
          class="css-mfolfb"
        >
          <a
            href="https://mc.commercetools.com/example-project/products/28a7442d/general"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="css-rm9mk1"
              color="muted"
              data-test-id="cf-ui-icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </a>
          <button
            aria-label="Delete"
            class="css-8mvp1w"
            data-test-id="cf-ui-icon-button"
            type="button"
          >
            <span
              class="css-k008qs"
            >
              <svg
                aria-hidden="true"
                class="css-1jrff5x"
                data-test-id="cf-ui-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`SortableListItem should render successfully the loading variation 1`] = `
<div>
  <article
    class="css-1bc2b9n"
    data-test-id="cf-ui-card"
  >
    <div
      class="css-1cjnmwb"
      data-card-part="wrapper"
    >
      <div
        class="css-161cbg"
        data-card-part="content"
      >
        <svg
          aria-label="Loading component..."
          class="css-1xwljqd"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip_5799)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient_7559);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip_5799"
            >
              <rect
                data-test-id="cf-ui-skeleton-image"
                height="48"
                rx="0"
                ry="0"
                width="48"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient_7559"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation_6859"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation_6859.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation_6859.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-kx63wx"
        >
          <img
            alt="Dress Twin-Set rose"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-1.jpg"
            style="display: none;"
          />
        </div>
        <section
          class="css-1phcpb1"
        >
          <h1
            class="css-19gqs9u"
            data-test-id="cf-ui-heading"
          >
            Dress Twin-Set rose
          </h1>
          <h3
            class="css-1dno5oq"
            data-test-id="cf-ui-subheading"
          >
            M0E20130820E90Z
          </h3>
        </section>
        <div
          class="css-mfolfb"
        >
          <a
            href="https://mc.commercetools.com/example-project/products/28a7442d/general"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="css-rm9mk1"
              color="muted"
              data-test-id="cf-ui-icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </a>
          <button
            aria-label="Delete"
            class="css-8mvp1w"
            data-test-id="cf-ui-icon-button"
            type="button"
          >
            <span
              class="css-k008qs"
            >
              <svg
                aria-hidden="true"
                class="css-1jrff5x"
                data-test-id="cf-ui-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`SortableListItem should render successfully the sortable variation 1`] = `
<div>
  <article
    class="css-1bc2b9n"
    data-test-id="cf-ui-card"
  >
    <div
      class="css-1cjnmwb"
      data-card-part="wrapper"
    >
      <div
        class="css-161cbg"
        data-card-part="content"
      >
        <div
          class="css-14jukvq"
        >
          <img
            alt="Dress Twin-Set rose"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-1.jpg"
            style="display: block;"
          />
        </div>
        <section
          class="css-1phcpb1"
        >
          <h1
            class="css-19gqs9u"
            data-test-id="cf-ui-heading"
          >
            Dress Twin-Set rose
          </h1>
          <h3
            class="css-1dno5oq"
            data-test-id="cf-ui-subheading"
          >
            M0E20130820E90Z
          </h3>
        </section>
        <div
          class="css-mfolfb"
        >
          <a
            href="https://mc.commercetools.com/example-project/products/28a7442d/general"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="css-rm9mk1"
              color="muted"
              data-test-id="cf-ui-icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </a>
          <button
            aria-label="Delete"
            class="css-8mvp1w"
            data-test-id="cf-ui-icon-button"
            type="button"
          >
            <span
              class="css-k008qs"
            >
              <svg
                aria-hidden="true"
                class="css-1jrff5x"
                data-test-id="cf-ui-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;
